### PR TITLE
Add hidden field to CourseRun model and index

### PR DESCRIPTION
**Pull request by rlucioni**
Also add a filter to the list view of CourseRun admin so that users can determine which course runs are hidden. ECOM-5687.

@edx/ecommerce

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -359,6 +359,7 @@ class CourseRun(TimeStampedModel):
     card_image_url = models.URLField(null=True, blank=True)
     video = models.ForeignKey(Video, default=None, null=True, blank=True)
     slug = models.CharField(max_length=255, blank=True, null=True, db_index=True)
+    hidden = models.BooleanField(default=False)
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/models.py:362]: Add `db_index=True`, and re-create the migration.

**Review by clintonb — APPROVED** [review]

**Review by schenedx — APPROVED** [review]
👍

File: course_discovery/apps/course_metadata/models.py (hunk 1 above)
Comment (cpennington) [reply course_discovery/apps/course_metadata/models.py:362]: An index on a boolean field may not actually be worth it... It's not going to have a whole lot of filtering value (unless we're frequently planning to query for the smaller of the subsets).

Comment (cpennington) [reply course_discovery/apps/course_metadata/models.py:362]: (See http://stackoverflow.com/questions/6084460/mysql-not-using-index-when-checking-1-but-using-it-with-0 for a fairly good summary of why you might or might not want to index this).

Comment (cpennington) [reply course_discovery/apps/course_metadata/models.py:362]: (If you've already discussed the tradeoffs, then a comment to that effect is probably in order)

Comment (rlucioni) [reply course_discovery/apps/course_metadata/models.py:362]: TIL, thanks @cpennington. I will revert my most recent change and remove the index.

**Pull request merged**